assets/js/modules/pagespeed-insights/components/DashboardPageSpeed.js

```diff
@@ -0,0 +1,137 @@
+import { __, _x, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import ProgressBar from '../../../components/progress-bar';
+import Link from '../../../components/link';
+import LabReportMetrics from './LabReportMetrics';
+import FieldReportMetrics from './FieldReportMetrics';
+import { STORE_NAME as CORE_FORMS } from '../../../googlesitekit/datastore/forms/constants';
+import { STORE_NAME as CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
+import {
+	STORE_NAME,
+	STRATEGY_MOBILE,
+	STRATEGY_DESKTOP,
+	DATA_SRC_FIELD,
+	DATA_SRC_LAB,
+	FORM_DASH_WIDGET,
+} from '../datastore/constants';
+import { sanitizeHTML } from '../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function DashboardPageSpeed() {
+	// TODO: remove legacy global fallback.
+	const permalink = global._googlesitekitLegacyData?.permaLink || global.googlesitekit?.permaLink;
+	const referenceURL = useSelect( ( select ) => select( CORE_SITE ).getReferenceSiteURL() );
+	const url = permalink || referenceURL;
+	const reportMobile = useSelect( ( select ) => select( STORE_NAME ).getReport( url, STRATEGY_MOBILE ) );
+	const reportDesktop = useSelect( ( select ) => select( STORE_NAME ).getReport( url, STRATEGY_DESKTOP ) );
+	const strategy = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_DASH_WIDGET, 'strategy' ) ) || STRATEGY_MOBILE;
+	const dataSrc = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_DASH_WIDGET, 'dataSrc' ) );
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	const setStrategyMobile = useCallback( () => setValues( FORM_DASH_WIDGET, { strategy: STRATEGY_MOBILE } ), [] );
+	const setStrategyDesktop = useCallback( () => setValues( FORM_DASH_WIDGET, { strategy: STRATEGY_DESKTOP } ), [] );
+	const setDataSrcField = useCallback( () => setValues( FORM_DASH_WIDGET, { dataSrc: DATA_SRC_FIELD } ), [] );
+	const setDataSrcLab = useCallback( () => setValues( FORM_DASH_WIDGET, { dataSrc: DATA_SRC_LAB } ), [] );
+
+	// Set the default data source based on report data.
+	useEffect( () => {
+		if ( ! reportMobile || ! reportDesktop ) {
+			return;
+		}
+		if ( reportMobile?.loadingExperience?.metrics && reportDesktop?.loadingExperience?.metrics ) {
+			setDataSrcField();
+		} else {
+			setDataSrcLab();
+		}
+	}, [ reportMobile, reportDesktop ] );
+
+	if ( ! reportMobile || ! reportDesktop || ! dataSrc ) {
+		return <ProgressBar />;
+	}
+
+	const reportData = strategy === STRATEGY_MOBILE ? reportMobile : reportDesktop;
+	const footerLinkHTML = sprintf(
+		/* translators: 1: link attributes, 2: translated service name */
+		__( 'Learn more details at <a %1$s>%2$s</a>', 'google-site-kit' ),
+		`href="${ addQueryArgs( 'https://developers.google.com/speed/pagespeed/insights/', { url } ) }" class="googlesitekit-cta-link googlesitekit-cta-link--external" target="_blank"`,
```

Comment:
Not sure if that's possible in combination with `sprintf` here, but could we somehow use the `Link` component instead and have that generate the HTML? Not a big deal, but the attributes here would pretty much be handled by that.

Reply:
I've done something similar with [`react-string-replace`](https://www.npmjs.com/package/react-string-replace) in the past. Not sure if we could do the same thing here since the output is translated but I agree, it's not the best, but a limitation of React.

Reply:
Hmm not too big of a deal, let's keep as is for now.